src/Microsoft.Graph.Core/Requests/BaseClient.cs

```diff
@@ -26,7 +26,7 @@ public class BaseClient : IBaseClient
         {
             this.BaseUrl = baseUrl;
             this.AuthenticationProvider = authenticationProvider;
```

Comment:
Do we need to hold onto the AuthenticationProvider here?  Could we just remove it from BaseClient?